unitTests/ui/FacetSearchTest.ts

```diff
@@ -144,23 +153,94 @@ export function FacetSearchTest() {
+
+              Simulate.keyUp($$(built).find('input'), KEYBOARD.DOWN_ARROW);
+              expect(getSearchResult(1).hasClass('coveo-facet-search-current-result')).toBe(true);
+              expect(getSearchResult(1).hasClass('coveo-facet-value-will-exclude')).toBe(false);
 
               Simulate.keyUp($$(built).find('input'), KEYBOARD.DOWN_ARROW);
-              expect($$($$(facetSearch.searchResults).findAll('li')[1]).hasClass('coveo-facet-search-current-result')).toBe(true);
+              expect(getSearchResult(1).hasClass('coveo-facet-search-current-result')).toBe(true);
+              expect(getSearchResult(1).hasClass('coveo-facet-value-will-exclude')).toBe(true);
 
               Simulate.keyUp($$(built).find('input'), KEYBOARD.DOWN_ARROW);
-              expect($$($$(facetSearch.searchResults).findAll('li')[2]).hasClass('coveo-facet-search-current-result')).toBe(true);
+              expect(getSearchResult(2).hasClass('coveo-facet-search-current-result')).toBe(true);
 
               Simulate.keyUp($$(built).find('input'), KEYBOARD.UP_ARROW);
-              expect($$($$(facetSearch.searchResults).findAll('li')[1]).hasClass('coveo-facet-search-current-result')).toBe(true);
+              expect(getSearchResult(1).hasClass('coveo-facet-search-current-result')).toBe(true);
+              expect(getSearchResult(1).hasClass('coveo-facet-value-will-exclude')).toBe(true);
 
               Simulate.keyUp($$(built).find('input'), KEYBOARD.UP_ARROW);
-              expect($$($$(facetSearch.searchResults).findAll('li')[0]).hasClass('coveo-facet-search-current-result')).toBe(true);
+              expect(getSearchResult(1).hasClass('coveo-facet-search-current-result')).toBe(true);
+              expect(getSearchResult(1).hasClass('coveo-facet-value-will-exclude')).toBe(false);
+
+              Simulate.keyUp($$(built).find('input'), KEYBOARD.UP_ARROW);
+              expect(getSearchResult(0).hasClass('coveo-facet-search-current-result')).toBe(true);
+              expect(getSearchResult(0).hasClass('coveo-facet-value-will-exclude')).toBe(true);
+
+              Simulate.keyUp($$(built).find('input'), KEYBOARD.UP_ARROW);
+              expect(getSearchResult(0).hasClass('coveo-facet-search-current-result')).toBe(true);
+              expect(getSearchResult(0).hasClass('coveo-facet-value-will-exclude')).toBe(false);
 
               // loop around !
               Simulate.keyUp($$(built).find('input'), KEYBOARD.UP_ARROW);
-              expect($$($$(facetSearch.searchResults).findAll('li')[9]).hasClass('coveo-facet-search-current-result')).toBe(true);
+              expect(getSearchResult(9).hasClass('coveo-facet-search-current-result')).toBe(true);
+              expect(getSearchResult(9).hasClass('coveo-facet-value-will-exclude')).toBe(true);
+              done();
+            });
+          });
+
+          function spyOnAndReturnFirstCheckbox() {
```

Comment:
Small nitpick, but i'd put all those utility test function at the top of the test block itself, instead of kind of in the middle.

Also `cons spyOnAndReturnFirstCheckbox = ()=> {...}`